wallet/core/src/storage/local/payload.rs

```diff
@@ -1,25 +1,28 @@
+//!
+//! Encrypted wallet payload storage.
+//!
+
 use crate::imports::*;
 use crate::result::Result;
 use crate::secret::Secret;
-use crate::storage::{Account, AddressBookEntry, PrvKeyData, PrvKeyDataId};
+use crate::storage::{AddressBookEntry, PrvKeyData, PrvKeyDataId};
 use kaspa_bip32::Mnemonic;
 use zeroize::{Zeroize, ZeroizeOnDrop};
 
-pub const PAYLOAD_VERSION: [u16; 3] = [1, 0, 0];
+const WALLET_PAYLOAD_MAGIC: u32 = 0x44415441;
+const WALLET_PAYLOAD_VERSION: u32 = 0;
```

Comment:
maybe make these variable trait, and implement trait for Payload?

Reply:
Not quite sure this is feasible.  Like first thought is to do this for Account variants, but Account variants can't deserialize as they need context to instantiate...  Hence `try_new` and factories that allow account to setup and then to deser.

Also, this is already a `BorshSerialize`/`BorshDeserialize` trait.  Magic numbers and versions are just gates really.  So not sure this is worth the effort right now.  We can look at refactoring this later, but for now I think this is ok as-is.

Reply:
I've renamed and scoped `STORAGE_VERSION` and `STORAGE_MAGIC` within associative structs that use them: https://github.com/aspectron/rusty-kaspa/commit/5d53f41702c80226d858e3eaacf3d935c47bf2c8

Reply:
So I made an effort yesterday and did a partial implementation of this.  I renamed `Storable` to `Payload` and turned `Storable` into a trait that carries `const` `MAGIC` and `VERSION` within itself.  So now we have `trait Storable : BorshSerialize + BorshDeserialize` which is passed directly to `AccountStorage` as a generic.

This applies to accounts only.  There is additional remodeling/improvement that can be done to the structure but at this point we should focus on that in the next stage of development.  Since everything is to a certain extent generalized/abstracted we can do in-house improvements and remodeling without affecting any external APIs.

I originally wanted to leave this alone but I made a mistake passing a wrong variable around that resulted in confusion.  At the same time passing `T: BorshSerialize` didn't help as almost all types have borsh impl and as such it does not prevent such errors.  Your suggestion to use a custom trait worked wonders as this constricts the type received by `AccountStorage` which significantly reduces the possibility of mishandling data in the future.

We can probably convert the trait to use its own serialization fns but for now I left it to use default Borsh trait fns.

Reply:
So we had lengthy discussions on the topic outside of github.  I am leaving all traits and related code as-is for the time being.  

I don't think traits are applicable here because the design goal of this is to provide explicit per-byte control of binary serialization to each struct in such a way that each struct has an explicit definition and control of deserialization, whereas creating a trait that manages this goes 100% against this design goal.  (this is akin to building an airport where all you need to do is to take 5 flights and since we had airport already I was trying to get rid of that and get 5 tickets).  

That said there are merits to this, and the discussions on the topic have shown other benefits related to trait use in relation to this (related to type constraints) that I have already integrated (and will make another push later today with some improvements). 

Leaving the current implementation as-is.